### PR TITLE
Audience is endpoint dependent.

### PR DESCRIPTION
**Pull request by rohe**
Thanks for contributing to this library! Please include the following check
list in your pull request submission (you can delete this message). If your
changes don't need a change log or documentation update, please ignore this.

- [ ] Any changes relevant to users are recorded in the `CHANGELOG.md`.
- [ ] The documentation has been updated, if necessary.

---

### CODE REVIEW

**Comment by rohe** [comment]
I really don't care if the coverage decreases with 0.001 % .

File: src/oic/utils/authn/client.py [hunk 1]
```diff
@@ -310,14 +311,16 @@ def construct(self, cis, request_args=None, http_args=None, **kwargs):
...
-        audience = self.cli.provider_info['issuer']
         algorithm = None
         if kwargs['authn_endpoint'] in ['token', 'refresh']:
             try:
                 algorithm = self.cli.registration_info[
                     'token_endpoint_auth_signing_alg']
             except (KeyError, AttributeError):
                 pass
+            audience = self.cli.provider_info['token_endpoint']
+        else:
```
Comment (tpazderka) [inline src/oic/utils/authn/client.py]: The drop in coverage is caused by this branch, which is never covered.

Comment (rohe) [reply src/oic/utils/authn/client.py]: Can easily be remedied

**Comment by rohe** [comment]
I don't think this will break anything. It's really a corner case, slightly outside the standard.
private_key_jwt and cleint_secret_jwt was designed for client authentication at the token endpoint.
But now when the methods are defined they are creatively :-) used at other endpoints.
In which case using the token endpoint URL as audience doesn't make sense.

**Pull request merged**